lib/wallet.py

```diff
@@ -2239,7 +2240,17 @@ def sign_transaction(self, tx, password, *, use_cache=False):
         for k in self.get_keystores():
             try:
                 if k.can_sign(tx):
-                    k.sign_transaction(tx, password, use_cache=use_cache)
+                    if ndata:  # used for reusable paycodes
+                        # test if keystore sign_transaction method knows about the `ndata` kwarg
+                        if 'ndata' in inspect.signature(k.sign_transaction, follow_wrapped=True).parameters:
+                            # keystore understands the optional `ndata` kwarg
+                            k.sign_transaction(tx, password, use_cache=use_cache, ndata=ndata)
+                        else:
+                            # keystore does not understand `ndata` (possibly because hw wallet)
+                            raise BaseException("Keystore does not understand ndata parameter.  Possibly wrong wallet type attemping special operation.")
```

Comment:
again, I recommend using `RuntimeError` here

Reply:
done in 060ca8f